### PR TITLE
Add logger for tracks events

### PR DESCRIPTION
**Pull request by joshuatf**
Adds logging to all WC tracks events.

cc @pmcpinto Is there any other information or features you would find useful here?

### Screenshots

Screen Shot 2022-01-27 at 11 33 41 AM
Screen Shot 2022-01-27 at 11 25 19 AM

### Testing

1. Install and activate this plugin
2. Make sure you've opted into tracking
3. Visit the WooCommerce->Status->Logs page `wp-admin/admin.php?page=wc-status&tab=logs`
4. Select a `tracks-*` log from the dropdown
5. Visit various pages to ensure tracks are fired

Note that if you want to trigger the task tracks events in the above photo, you will need to delete the `woocommerce_task_list_tracked_completed_tasks` to retrigger them.

### CODE REVIEW

**Comment by pmcpinto** [comment]
Thanks for the ping. I think the information included is sufficient

File: api/tracks/tracks-debug-log.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	private $source = 'tracks';
+
+	/**
+	 * Initialize hooks.
+	 */
+	public function __construct() {
+		add_filter( 'woocommerce_tracks_event_properties', array( $this, 'log_event' ), 10, 2 );
+		$logger = wc_get_logger();
+		$this->logger = $logger;
+		$this->logger = $logger;
```
Comment (octaedro) [inline api/tracks/tracks-debug-log.php:28]: It seems like this line is duplicated.

**Review by octaedro — APPROVED** [review]
Nice job @joshuatf! I just left a small comment. This is testing well on my end and other than the comment I left the code looks good. I'm going to pre-approve this PR.

**Comment by mattsherman** [comment]
@joshuatf Why are you holding out on us all and not merging in this hotness? 😄

**Comment by mattsherman** [comment]
@joshuatf Let's get this merged.

**Pull request merged**